### PR TITLE
Add tryGettingSimpleWorkflowResult and tryGettingComplexWorkflowResult

### PR DESCRIPTION
**Pull request by zklgame**
issue: https://github.com/indeedeng/iwf-java-sdk/issues/85

### CODE REVIEW

File: src/main/java/io/iworkflow/core/Client.java [hunk 1]
```diff
@@ -239,44 +239,121 @@ private List<SearchAttribute> convertToSearchAttributeList(final Map<String, Sea
...
+    /**
+     * In some cases, a workflow may have more than one completion states.
+     * If the workflow is still running, return an empty list.
+     * Else, return the same result as {@link #getComplexWorkflowResultWithWait(String, String)}.
+     *
+     * @param workflowId    required, the workflowId
+     * @param workflowRunId optional, can be empty
+     * @return a list of the state output for completion states. User code will figure how to use ObjectEncoder to decode the output
+     */
+    public List<StateCompletionOutput> getComplexWorkflowResultWithoutWait(
```
Comment (longquanzheng) [inline src/main/java/io/iworkflow/core/Client.java:340]: I personally want to avoid using `WithoutWait` because it looks too similar to `WithWait` at quick glance... 
If you look at 
* getComplexWorkflowResultWithoutWait
* getComplexWorkflowResultWithWait
The `out` is too minor for our human eyes. 

Though it looks accurate in grammar 😮‍💨 

Maybe just call tryGettingComplexWorkflowResult -- it seems like Temporal is going to have this name as well: https://github.com/temporalio/features/issues/314

File: src/main/java/io/iworkflow/core/UnregisteredClient.java [hunk 2]
```diff
@@ -209,6 +306,33 @@ public <T> T getSimpleWorkflowResultWithWait(
         return clientOptions.getObjectEncoder().decode(output.getCompletedStateOutput(), valueClass);
     }
 
+    private WorkflowGetResponse getWorkflowResponse(
```
Comment (longquanzheng) [inline src/main/java/io/iworkflow/core/UnregisteredClient.java:309]: Nit: to me this should be called `getWorkflowResult`, and the other one could be something like `getWorkflowResultAndDecode`

File: src/main/java/io/iworkflow/core/Client.java (hunk 1 above)
Comment (zklgame) [reply src/main/java/io/iworkflow/core/Client.java:340]: Good call, yeah when I was writting the codes I also needed to pay attention on the names...

File: src/main/java/io/iworkflow/core/UnregisteredClient.java (hunk 2 above)
Comment (zklgame) [reply src/main/java/io/iworkflow/core/UnregisteredClient.java]: I used the plural form `getWorkflowResults` because the return values is from `workflowGetResponse.getResults()`.

File: src/main/java/io/iworkflow/core/Client.java [hunk 3]
```diff
@@ -239,44 +239,121 @@ private List<SearchAttribute> convertToSearchAttributeList(final Map<String, Sea
...
+            final Class<T> valueClass,
             final String workflowId) {
         return getSimpleWorkflowResultWithWait(valueClass, workflowId, "");
     }
 
     /**
-     * In some cases, a workflow may have more than one completion states
+     * For most cases, a workflow only has one result(one completion state).
+     * Use this API to retrieve the output of the state.
+     * If the workflow is still running, return null.
```
Comment (longquanzheng) [inline src/main/java/io/iworkflow/core/Client.java:276]: Returnning null will be a problem if some workflow actually return null as result. There will be hard to differentiate the null being results or workflow uncompleted.

File: src/main/java/io/iworkflow/core/UnregisteredClient.java [hunk 4]
```diff
@@ -162,53 +163,176 @@ public String startWorkflow(
...
+            if (withWait) {
+                workflowGetResponse = defaultApi.apiV1WorkflowGetWithWaitPost(request);
+            } else {
+                workflowGetResponse = defaultApi.apiV1WorkflowGetPost(request);
+            }
+        } catch (final FeignException.FeignClientException exp) {
+            throw IwfHttpException.fromFeignException(clientOptions.getObjectEncoder(), exp);
+        }
+
+        if (withWait && workflowGetResponse.getWorkflowStatus() != WorkflowStatus.COMPLETED) {
```
Comment (longquanzheng) [inline src/main/java/io/iworkflow/core/UnregisteredClient.java:325]: so maybe just throw uncompletedException here when withWait==false and workflow is not completed

File: src/main/java/io/iworkflow/core/Client.java [hunk 5]
```diff
@@ -239,44 +239,129 @@ private List<SearchAttribute> convertToSearchAttributeList(final Map<String, Sea
     }
 
     /**
-     * For most cases, a workflow only has one result(one completion state)
-     * Use this API to retrieve the output of the state
+     * For most cases, a workflow only has one result(one completion state).
+     * Use this API to retrieve the output of the state with waiting for the workflow to complete.
+     * If the workflow is not COMPLETED, throw the {@link WorkflowUncompletedException}.
```
Comment (longquanzheng) [inline src/main/java/io/iworkflow/core/Client.java:244]: This is great. Can you also help add the similar comment to another APIs which will throw a different exception today https://github.com/indeedeng/iwf-java-sdk/issues/202 ( you can do it in a different PR)

Comment (zklgame) [reply src/main/java/io/iworkflow/core/Client.java]: True...

**Review by longquanzheng — APPROVED** [review]
